torch/testing/_internal/common_utils.py

```diff
@@ -933,13 +933,23 @@ def __torch_function__(self, func, types, args=(), kwargs=None):
 
 def skipIfTorchDynamo(msg="test doesn't currently work with torchdynamo"):
     def decorator(fn):
-        @wraps(fn)
-        def wrapper(*args, **kwargs):
-            if TEST_WITH_TORCHDYNAMO:
-                raise unittest.SkipTest(msg)
-            else:
-                fn(*args, **kwargs)
-        return wrapper
+        if not isinstance(fn, type):
+            @wraps(fn)
+            def wrapper(*args, **kwargs):
+                if TEST_WITH_TORCHDYNAMO:
+                    raise unittest.SkipTest(msg)
+                else:
+                    fn(*args, **kwargs)
+            return wrapper
+
+        assert(isinstance(fn, type))
+        if TEST_WITH_TORCHDYNAMO:
+            fn.__unittest_skip__ = True
+            fn.__unittest_skip_why__ = msg
```

Comment:
You know that you can just reuse the unittest.skip combinator here right

Reply:
I know, I was concerned someone would use `skipIfTorchDynamo` on a class again.